nestjs/src/courses/students/students.service.ts

```diff
@@ -25,29 +53,55 @@ export class StudentsService {
     return student;
   }
 
-  public async addStudentToTeamDistribution(studentId: number, teamDistribution: TeamDistribution) {
-    const student = await this.getStudentWithTeamDistributions(studentId);
+  public async getStudentDetailed(studentId: number) {
+    const student = await this.studentRepository.findOneOrFail({
+      where: { id: studentId },
+      relations: ['teams', 'teamDistribution', 'user'],
+    });
+    return student;
+  }
+
+  public async addStudentToTeamDistribution(
+    studentId: number,
+    teamDistribution: TeamDistribution,
+    withVerification = true,
+  ) {
+    const student = await this.getStudentDetailed(studentId);
     const currentDate = dayjs();
     const distributionStartDate = dayjs(teamDistribution.startDate);
     const distributionEndDate = dayjs(teamDistribution.endDate);
-    if (student == null) throw new NotFoundException();
-    if (currentDate < distributionStartDate || currentDate > distributionEndDate) {
+    if (withVerification && (currentDate < distributionStartDate || currentDate > distributionEndDate)) {
       throw new BadRequestException();
     }
-    if (student.totalScore < teamDistribution.minTotalScore) {
+    if (withVerification && student.totalScore < teamDistribution.minTotalScore) {
       throw new BadRequestException('Number of points is less than the input threshold for distribution');
     }
+    if (withVerification && student.teams.find(el => el.teamDistributionId === teamDistribution.id)) {
+      throw new BadRequestException();
+    }
     student.teamDistribution = [...student.teamDistribution, teamDistribution];
     await this.studentRepository.save(student);
   }
 
+  public async addStudentToTeam(studentId: number, team: Team) {
+    const student = await this.getStudentDetailed(studentId);
+    student.teams = [...student.teams, team];
```

Comment:
won't it be better to push `team` into `teams`?

Reply:
fixed